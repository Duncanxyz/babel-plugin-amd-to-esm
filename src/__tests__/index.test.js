const babel = require('@babel/core');
const plugin = require('../index.js');

const fs = require('fs');
const path = require('path');

function makeTest(name) {
  const originCode = fs.readFileSync(
    path.resolve(__dirname, `./code/${name}.js`),
    'utf8'
  );

  const { code } = babel.transform(originCode, { plugins: [plugin] });
  expect(code).toMatchSnapshot();
}

it('all feature', () => {
  makeTest('all-feature');
});

it('no dependencies two params', () => {
  makeTest('no-dependencies-two-params');
});

it('no dependencies one param', () => {
  makeTest('no-dependencies-one-param');
});

it('arrow function', () => {
  makeTest('arrow-function');
});
