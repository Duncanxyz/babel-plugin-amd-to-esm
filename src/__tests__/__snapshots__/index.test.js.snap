// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all feature 1`] = `
"console.log('before code');
import a from "depA";
import b from "depB";
import _amdDep from "depC";
import _amdDep2 from "depD";
import("depE");
const {
  c1,
  c2,
  c3: c3x,
  c4: {
    cc4: {
      ccc4
    }
  }
} = _amdDep;
const {
  d1
} = _amdDep2;
// inner code
console.log('inner code start');
export default {
  property: 'property',
  name: 123
};
console.log('inner code end');
console.log('after code');"
`;

exports[`arrow function 1`] = `
"import a from "a";
export default {
  name: 'module-arrow-function'
};"
`;

exports[`no dependencies one param 1`] = `
"export default {
  name: 'module-no-dependencies-one-param'
};"
`;

exports[`no dependencies two params 1`] = `
"export default {
  name: 'module-no-dependencies-two-params'
};"
`;
